main.tf

```
@@ -298,7 +334,7 @@ resource "aws_kms_alias" "this" {
 
   name          = var.aliases_use_name_prefix ? null : "alias/${each.value.name}"
   name_prefix   = var.aliases_use_name_prefix ? "alias/${each.value.name}-" : null
-  target_key_id = var.create_external ? aws_kms_external_key.this[0].id : aws_kms_key.this[0].key_id
+  target_key_id = var.create_external ? aws_kms_external_key.this[0].id : (var.create_replica_external ? aws_kms_replica_external_key.this[0].id : (var.create_replica ? aws_kms_replica_key.this[0].key_id : aws_kms_key.this[0].key_id))
```

I think we can simplify this to simply
```suggestion
  target_key_id = try(aws_kms_key.this[0].key_id, aws_kms_external_key.this[0].id, aws_kms_replica_key.this[0].key_id, aws_kms_replica_external_key.this[0].id)
```

since you are guarding against more than one key being created, only one in that array will resolve

---

Thanks @bryantbiggs, I've updated these and added outputs.